auditlog/audit_log_blackbox_test.go

```diff
@@ -0,0 +1,205 @@
+	testsuite "github.com/fabric8-services/fabric8-common/test/suite"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RepositoryBlackBoxTest struct {
+	testsuite.DBTestSuite
+	repo auditlog.Repository
+}
+
+func TestRecordRepository(t *testing.T) {
+	resource.Require(t, resource.Database)
+	config, err := configuration.New()
+	require.NoError(t, err)
+	suite.Run(t, &RepositoryBlackBoxTest{DBTestSuite: testsuite.NewDBTestSuite(config)})
+}
+
+func (s *RepositoryBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.repo = auditlog.NewRepository(s.DB)
+}
+
+func (s *RepositoryBlackBoxTest) TestCreateRecord() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		before := time.Now()
+		auditLog := auditlog.AuditLog{
+			EventTypeID: auditlog.UserSearch,
+			IdentityID:  uuid.NewV4(),
+			EventParams: auditlog.EventParams{},
+		}
+		// when
+		err := s.repo.Create(context.Background(), &auditLog)
+		// then
+		require.NoError(t, err)
+		assert.NotEqual(t, uuid.NullUUID{}, auditLog.ID)
+		assert.True(t, auditLog.CreatedAt.After(before)) // "is after before". hahahaha....
+	})
+
+	s.T().Run("failure", func(t *testing.T) {
+
+		t.Run("missing event type", func(t *testing.T) {
+			// given
+			auditLog := auditlog.AuditLog{
+				IdentityID:  uuid.NewV4(),
+				EventParams: auditlog.EventParams{},
+			}
+			// when
+			err := s.repo.Create(context.Background(), &auditLog)
+			// then
+			require.Error(t, err)
+		})
+
+		t.Run("missing identity id", func(t *testing.T) {
+			// given
+			auditLog := auditlog.AuditLog{
+				EventTypeID: auditlog.UserSearch,
+				EventParams: auditlog.EventParams{},
+			}
+			// when
+			err := s.repo.Create(context.Background(), &auditLog)
+			// then
+			require.Error(t, err)
+		})
+	})
+}
+
+func (s *RepositoryBlackBoxTest) TestLoadByID() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		auditLog := auditlog.AuditLog{
+			EventTypeID: auditlog.UserSearch,
+			IdentityID:  uuid.NewV4(),
+			EventParams: auditlog.EventParams{
+				"idx": 1,
+			},
+		}
+		err := s.repo.Create(context.Background(), &auditLog)
+		require.NoError(t, err)
+		// when
+		result, err := s.repo.LoadByID(context.Background(), auditLog.ID)
+		// then
+		require.NoError(t, err)
+		// comparing 'CreatedAt' may cause troubles b/c of nanosecond roundings, so let's just verify that the result ID is the one expected
+		assert.Equal(t, auditLog.ID, result.ID)
+
+	})
+	s.T().Run("not found", func(t *testing.T) {
+		// when
+		_, err := s.repo.LoadByID(context.Background(), uuid.NewV4())
+		// then
+		require.Error(t, err)
+		assert.IsType(t, errors.NotFoundError{}, err)
+	})
+}
+
+func (s *RepositoryBlackBoxTest) TestList() {
+	// given 2 users with 12 auditLogs each
+	identity1 := uuid.NewV4()
+	identity2 := uuid.NewV4()
+	for _, identity := range []uuid.UUID{identity1, identity2} {
```

Comment:
Or maybe worth adding TestGrpah?
cc: @sbryzak

Reply:
I don't believe it's worth it for now, but if the schema grows then yes, why not.